solutions/fully-configurable/main.tf

```diff
@@ -32,3 +41,76 @@ module "scc_wp" {
   scc_wp_service_plan           = var.scc_workload_protection_service_plan
   cbr_rules                     = var.cbr_rules
 }
+
+########################################################################################################################
+# Beginning of Cloud Security Posture Management (CSPM) Configuration
+########################################################################################################################
+# SCC Workload Protection Trusted Profile
+########################################################################################################################
+
+# Create Trusted profile for SCC Workload Protection instance
+module "trusted_profile_scc_wp" {
+  count                       = var.cspm_enabled ? 1 : 0
+  source                      = "terraform-ibm-modules/trusted-profile/ibm"
+  version                     = "2.1.1"
+  trusted_profile_name        = local.scc_workload_protection_trusted_profile_name
+  trusted_profile_description = "Trusted Profile for SCC workload Protection instance ${module.scc_wp.guid} with required access for configuration aggregator."
+
+  trusted_profile_identity = {
+    identifier    = module.scc_wp.crn
+    identity_type = "crn"
+  }
+
+  trusted_profile_policies = [
+    {
+      roles = ["Service Configuration Reader", "Viewer", "Configuration Aggregator Reader"]
+      resources = [{
+        service = "apprapp"
+      }]
+      description = "App Config access"
+    },
+  ]
+
+  trusted_profile_links = [{
+    cr_type = "VSI"
+    links = [{
+      crn = module.scc_wp.crn
+    }]
+  }]
+}
+
+################################################################
+# IAM Auth Token
+################################################################
+
+data "ibm_iam_auth_token" "auth_token" {
+  depends_on = [module.trusted_profile_scc_wp]
+}
+
+################################################################
+# Enable CSPM for SCC Workload Protection instance
+################################################################
+
+# CSPM can only be enabled after the trusted profile exists,
+# but profile can only exist after instance has been created
+# hence we cannot directly enable CSPM in the instance creation
+# and need to use a separate resource to enable it
+resource "restapi_object" "enable_cspm" {
```

Comment:
This should go into the root level module

Reply:
It means the usage block in the main readme will need to be updated with the required restapi provider config